mustang_codegen/lib/src/state_generator/screen_state_generator.dart

```diff
@@ -49,51 +53,75 @@ class ScreenStateGenerator extends Generator {
     );
 
     return '''
-      import 'dart:convert';
       import 'dart:developer';
       import 'package:flutter/foundation.dart';
+      import 'package:flutter/widgets.dart';
       import 'package:mustang_core/mustang_core.dart';
+      import 'package:mustang_core/mustang_widgets.dart';
       ${stateImports.join('\n')}
       
-      class $stateName extends ChangeNotifier {
-        $stateName() {
-          mounted = true;
+      $importGenService
+      
+      class $stateName extends ChangeNotifier implements RouteAware {
```

Comment:
have you tested this by adding a print statement to the screen? in my testing yesterday the UI was rapidly rebuilt with `RouteAware`. I didn't investigate it but better to check it now than revert changes later on

Reply:
That was the bug in the events project, fixed in the examples